src/filemap.rs

```diff
@@ -36,16 +36,95 @@ pub fn write_all_files(file_map: &FileMap,
                        config: &Config)
                        -> Result<(HashMap<String, String>), io::Error> {
     let mut result = HashMap::new();
+    output_heading(mode).ok();
+
     for filename in file_map.keys() {
         let one_result = try!(write_file(&file_map[filename], filename, mode, config));
         if let Some(r) = one_result {
             result.insert(filename.clone(), r);
         }
     }
+    output_trailing(mode).ok();
 
     Ok(result)
 }
 
+pub fn output_heading(mode: WriteMode) -> Result<(), io::Error> {
+    let stdout = stdout();
+    let mut stdout = stdout.lock();
+    match mode {
+        WriteMode::Checkstyle => {
+            let mut xml_heading = String::new();
+            xml_heading.push_str("<?xml version=\"1.0\" encoding=\"utf-8\"?>");
+            xml_heading.push_str("\n");
+            xml_heading.push_str("<checkstyle version=\"4.3\">");
+            try!(write!(stdout, "{}", xml_heading));
+            Ok(())
+        }
+        _ => Ok(()),
+    }
+}
+
+pub fn output_trailing(mode: WriteMode) -> Result<(), io::Error> {
+    let stdout = stdout();
+    let mut stdout = stdout.lock();
+    match mode {
+        WriteMode::Checkstyle => {
+            let mut xml_tail = String::new();
+            xml_tail.push_str("</checkstyle>");
+            try!(write!(stdout, "{}", xml_tail));
+            Ok(())
+        }
+        _ => Ok(()),
+    }
+}
+
+pub fn output_checkstyle_file<T>(mut writer: T,
+                                 filename: &str,
+                                 diff: Vec<Mismatch>)
+                                 -> Result<(), io::Error>
+    where T: Write
+{
+    try!(write!(writer, "<file name=\"{}\">", filename));
+    for mismatch in diff {
+        for line in mismatch.lines {
+            match line {
+                DiffLine::Expected(ref str) => {
+                    let message = xml_escape_str(&str);
+                    // TODO XML encode str here.
+                    try!(write!(writer,
+                                "<error line=\"{}\" severity=\"error\" message=\"Should be \
```

Comment:
I think so, formatting errors are not compiler's errors, so I would prefer my tools to not show those red underlines to me :D